### ISSUE TITLE
Help needed to prepare the 0.300 release

### ISSUE THREAD

**Issue by sdwolfz** [syl20bnr/spacemacs#11741]
In order to prepare the next release we need to have all the commits merged in since the last release summarized into the changelog. This process is a one time thing, as we plan to integrate it as part of our merge process in the future, but now it needs to be done manually.

We started working towards this on the [release_notes_0.300](https://github.com/syl20bnr/spacemacs/tree/release_notes_0.300) branch. You can see an example in this commit: https://github.com/syl20bnr/spacemacs/commit/8cf612771bf494b2249bda68800a80e948fd194b.

There are `1418` commits left to summarize, about one and a half year's worth of merges. How we've done it so far was to pin a specific commit (69a89cc41243cfb189238a2297fde4498d7fcf6d) and navigate the github log pages until the first commit that was not summarized. Currently we are at this page: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+1418

Notice the query parameter in the url: `before=69a89cc41243cfb189238a2297fde4498d7fcf6d` tells it from which commit to start pagination and `+1418` tells it how many commits we've progressed. There are 35 commits per page and it takes us 30-40 minutes to summarize one page, depending on what the commits changed.

As there are not many of us able to work on this, and because it's taking a long time, we were thinking of crowd-sourcing this activity in the following way:

- At the bottom of this post there will be a list of all the pages not yet summarized.
- Reserve one page per person at a time, starting from top to bottom. Please do not skip pages as it makes it more difficult to track.
- Summarize all the commits on that page in the `CHANGELOG.develop` file and open a PR against the `release_notes_0.300` branch.
- You can reserve another page while waiting to get merged but open a separate PR with it. It's easier to review this way.
- One of the maintainers will assign themselves to that PR, review and merge it.
- Please specify which page and from which commit you start in the body and add a pointer to the next page and commit:
    ```
    Summarizing: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+1418
    Starting with: cb17b2773badb4a86e0ffc1b366e5b8589cbea66
    
    New pointer: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+1383
    Start with: 106f98497ea91730cb53ece1733d87bf7fae2e51
    ```
- Don't forget to add your name in the `Release notes summarized` section if you wish to.
- If you can not finish a page in the same day, please release it so that somebody else can pick it up.
- If it's unclear where exactly to put something, ask about it in the PR. Here are some guidelines:
    * Any breaking change goes in `*** Breaking Changes`.
    * Changes to the core (the `core/` directory) go in the `*** Core changes` section.
    * Changes to the spacemacs distributions and spacemacs layers go in the `*** Distribution changes` section.
    * Documentation fixes, code style changes and CI are grouped at the end.
    * Changes to a newly added layer should not be added in a new entry, instead put the contributor name together with the person that added the layer.
    * Everything else should be under it's specific layer.
    * Look at the previously summarized commits if you want to see an example of a specific element.
- For each change you can use the commit title as a summary, but make sure to edit it in case it does not make sense, or if you need to add extra/less detail.
- At the end of the summary add `(thanks to ` and the person's name. Make sure to use the person's real name if available, otherwise use the github username. If all else fails, there will be a name in the commit itself that you can use.
- If you notice that a commit builds on top of an existing one you can group those together.
- If one of the maintainers (@syl20bnr @bmag @d12frosted @TheBB @JAremko @sdwolfz) did a trivial change to one of the already summarized commits, like adding documentation, code formatting or moving stuff around, you can skip over it, but not if it was done by a contributor.
- It is not necessary to mention @syl20bnr

How to reserve:
- Look in the list below for the first open entry.
- Search for it in this thread and see if it was not reserved in the last 24 hours and it does not have an associated PR open.
- Announce that you reserve it by posting:
    ```
    Reserving:  https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+1418
    ```
- Once you open a PR edit your post to say:
    ```
    Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+1418
    PR: #8550 
    ```

Here is a list of all the pages, the ticked ones are already done `40/40`:
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+1418
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+1383
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+1348
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+1313
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+1278
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+1243
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+1208
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+1173
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+1138
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+1103
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+1068
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+1033
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+998
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+963
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+928
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+893
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+858
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+823
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+788
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+753
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+718
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+683
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+648
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+613
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+578
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+543
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+508
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+473
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+438
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+403
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+368
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+333
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+298
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+263
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+228
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+193
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+158
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+123
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+88
- [x] https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+53

Note that these are not actually all the commits, there are still some left on the develop branch. We will take care of those at the end. Also there are some bug fixes that we'll need to do just before merging, so we'll end up making sure everything works for a few more days after we finish this mammoth of a task.

Happy hacking and thank you for the help!

--------

_edited by @syl20bnr to mention that it is not necessary to credit Sylvain Benner_

--------
Thank you everybody for participating in this :heart:! Your effort made a huge difference as this would have taken us 2-3 more months to finish on our own. Now we can focus on all the remaining regressions until we release as scheduled at the end of the month.

**Comment by sdwolfz** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+1418

**Comment by sdwolfz** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+1383

**Comment by sdwolfz** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+1313

**Comment by sdwolfz** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+1278

**Comment by beyondpie** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+1208

**Comment by sdwolfz** [comment]
Thanks for joining in @beyondpie. 

I just noticed I got some instructions wrong, the ones about the "new pointer" and "start with". Will update the main post with fixes and clarifications.

**Comment by sdwolfz** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+1243

**Comment by randomwangran** [comment]
As a newbie to teamwork, I want to try my best.

Reserving:    https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+1173

**Comment by sdwolfz** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+1138

**Comment by beyondpie** [comment]
@sdwolfz 
I see 1173 haven't beed merged before your merged 1134.  
Does it mean that we can reserve some link after the link hasn't  been merged but reserved?  For example, I can now reserve 1103, and others can then reserve 1068 after my reserving though it hasn't been merged.

**Comment by igoralmeida** [comment]
@beyondpie I was taking a look at 1103 since a couple of minutes ago, intending to reserve it. Maybe skip it and take 1068 instead?
I think PR reviewers will take care of updates done in later pages, so one can just focus on summarizing their page.

**Comment by igoralmeida** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+1103

**Comment by sdwolfz** [comment]
@beyondpie yes you can reserve a page before the previous one was merged, and I'll try to consolidate stuff during reviews. Just don't intentionally skip over pages.

While this does not necessarily make stuff faster in terms of time, as I still have to spend time for reviews it does help alot in terms of effort as it's easyer to move things around, delete, or fix minor formatting issues compared to doing a page from scratch. Every bit helps.

**Comment by igoralmeida** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+1068

**Comment by gcentauri** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+998

**Comment by yuhan0** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+963

**Comment by igoralmeida** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+928

**Comment by sdwolfz** [comment]
@randomwangran It's been 4 days since you reserved [+1173](https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+1173) and you did not open a PR yet. You have some commits on your [release_notes_0.300](https://github.com/randomwangran/spacemacs/commits/release_notes_0.300) branch but at this rate it will take a long time to finish it. 

I will do that page myself now to be able to continue with merging the other open PRs.

Please don't feel upset about it, this is a boring and tedious task which consumes a lot of time and we should not turn it into a bigger blocker than it is right now. Feel free to pick up a new page but only if you feel confident you can finish it in a day, otherwise your time will be wasted.

**Comment by sdwolfz** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+1173
(Stolen from @randomwangran due to time)

**Comment by yuhan0** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+892

**Comment by gcentauri** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+858

**Comment by randomwangran** [comment]
> Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+1173
> (Stolen from @randomwangran due to time)
> Merged: [52a7454](https://github.com/syl20bnr/spacemacs/commit/52a7454e0b8af76f84239dcd1e017c279c092aa1)


I am sooooooo sad..

I did not see any information ( about due time ) when I decide to help. Indeed, if you point out this information, I would not have a try because I am totally newbiew to working as a team member.

I spend a lot of time (3~4) hours to read the document, to see how other people works, how the documents looks like, how to wirte a good summary to the commit message...

 Actually, I see a lots of items still not being processed. So, I think I am okay before, you guys finish the whole project.

Actually, as the time passby, I become more efficient to summary those items...

I would definite want to help, but if the condition is "finish it within oneday", I am afraid that I cannot do it.

**Comment by sdwolfz** [comment]
@randomwangran I have this bullet point in the original post:
> If you can not finish a page in the same day, please release it so that somebody else can pick it up.

You can still help by reviewing the already opened PRs, as you did with #11766.

**Comment by gcentauri** [comment]
@randomwangran - it does say to please release it if you cannot finish it in one day.  i think this still could be a great option to try and work as a team member!  maybe now that you're more familiar with it, you can try to reserve another down the line and get it done?

**Comment by sdwolfz** [comment]
@gcentauri In that case remove the line from the changelog. The end result is what the changelog should reflect.

**Comment by yuhan0** [comment]
I think that would be a good idea for the "Various Improvements" section at the end , we should just add new entries as a new line each and make it easier on the maintainers (fewer merge conflicts, just sort-lines and fill paragraph at the end of everything)

**Comment by Miciah** [comment]
Reserving:￼ https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+823

**Comment by Compro-Prasad** [comment]
I would also like each name to be on a separate line. This will help finding names easier.

**Comment by Compro-Prasad** [comment]
Would like to reserve: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+788

**Comment by Compro-Prasad** [comment]
Should I create a `cc-mode` header or will it go under `C-C++` header?

**Comment by sdwolfz** [comment]
@Compro-Prasad It's in ` layers/+lang/c-c++/packages.el` so it should go in the `C-C++` section.

**Comment by Compro-Prasad** [comment]
Do all the doc changes go under `Various improvements` header?

**Comment by Miciah** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+753

**Comment by Miciah** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+718

**Comment by syl20bnr** [comment]
Wonderful work!
This is not necessary to mention me in the release notes.

**Comment by syl20bnr** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+683

**Comment by Compro-Prasad** [comment]
I don't know, but there is a chance that maintainers may change.

**Comment by Miciah** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+648

**Comment by testandset** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+613

**Comment by Miciah** [comment]
Many new layers were extracted from existing layers, and I realize now that in addition to adding an entry under "New layers", I probably should add an entry under "Breaking Changes/x" when I extract some layer y from layer x.  Does that seem reasonable? If so, I can review the list of new layers and add any needed entries under "Breaking Changes".

**Comment by Miciah** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+543

**Comment by randomwangran** [comment]
@sdwolfz I found a newbie like me. He seems not notice the point where I also did not notice.

> It's been 4 days since you reserved +1173 and you did not open a PR yet.


[Please help him just like what you did to me : )](https://github.com/syl20bnr/spacemacs/issues/11741#issuecomment-451269820)

**Comment by sdwolfz** [comment]
@randomwangran Yup, I will steal his reservation once I am home and start working on this again... unless somebody else does it.

**Comment by sdwolfz** [comment]
@syl20bnr Unfortunately you did not finish https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+683 fast enough so I summarized it instead in order to unblock the queue.
Commit: d083b17b614afd22cfe17e58108d4b63677e96cc

**Comment by sdwolfz** [comment]
Reserving:  https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+508

**Comment by Miciah** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+473

**Comment by Miciah** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+438

**Comment by deb0ch** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+403

**Comment by JHJ93** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+368

**Comment by gcentauri** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+333

**Comment by yuhan0** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+298

**Comment by yuhan0** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+263

**Comment by sdwolfz** [comment]
@deb0ch you did not finish summarizing https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+403 in time so I did it.
Commit: e5d2cb5565b4e70aaafbb9039b8b17b95bbd9f52

**Comment by sdwolfz** [comment]
Thank you :heart:!
Cherry-picked into `release_notes_0.300` branch, you can safely delete your branch.
Commit: ad22ad0b64e7d092d637c5df98278ac2201a172d

**Comment by sdwolfz** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+228

**Comment by sdwolfz** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+193

**Comment by sdwolfz** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+158

**Comment by sdwolfz** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+123

**Comment by sdwolfz** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+88

**Comment by Miciah** [comment]
I made https://github.com/syl20bnr/spacemacs/pull/11833 with some fixes, including adding entries under "Breaking Changes" for functionality that has been extracted from existing layers into new layers.

**Comment by sdwolfz** [comment]
Reserving: https://github.com/syl20bnr/spacemacs/commits/develop?before=69a89cc41243cfb189238a2297fde4498d7fcf6d+53

**Comment by sdwolfz** [comment]
Thank you everybody for participating and making this release a reality :heart:!

We can now focus on the remaining regressions and we should be able to release at the end of the month.

**Comment by deb0ch** [comment]
@sdwolfz I was sure I would have time that day but something came up and today is too late 😅 

Sorry for the noise
